x-pack/plugins/enterprise_search/public/applications/shared/api_logic/create_api_logic.ts

```diff
@@ -23,19 +24,46 @@ export interface Actions<Args, Result> {
   makeRequest(args: Args): Args;
 }
 
+export interface CreateApiOptions<Result> {
+  clearFlashMessagesOnMakeRequest: boolean;
+  showErrorFlash: boolean;
+  showSuccessFlashFn?: (result: Result) => string;
```

Comment:
I could see use for adding a handler for the error function too, but I don't think we should add it until we find we need it for something specifically.

Reply:
Fully agreed, that's why I ended up not adding it